src/MacroTools/Buffs/DelayedRecallBuff.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using MacroTools.Extensions;
+using WCSharp.Buffs;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace MacroTools.Buffs
+{
+  public sealed class DelayedRecallBuff : PassiveBuff
+  {
+    /// <summary>
+    /// The units that should be moved when the buff is disposed off.
+    /// </summary>
+    private List<unit> UnitsToMove { get;}
+    
+    /// <summary>
+    /// The duration of the buff when first cast.
+    /// </summary>
+    public int InitialDuration { get; init; }
+    
+    /// <summary>
+    /// The point of the casting unit when first cast.
+    /// </summary>
+    private Point TargetPosition { get; }
+    
+    /// <summary>
+    /// The effect that is created when the buff is applied.
+    /// </summary>
+    private new effect? Effect { get; set; }
+
+    /// <inheritdoc />
+    public DelayedRecallBuff(unit caster, unit target, List<unit> unitsToMove) : base(caster, target)
+    {
+      UnitsToMove = unitsToMove;
+      TargetPosition =  new Point(GetUnitX(caster), GetUnitY(caster));
+    }
+    
+    public override void OnApply()
+    {
+      foreach (var unit in UnitsToMove)
+      {
+        unit.Show(false).SetInvulnerable(true);
+      }
+      
+      Effect = AddSpecialEffect(@"Abilities\Spells\Undead\Darksummoning\DarkSummonTarget.mdl", TargetPosition.X, TargetPosition.Y)
+        .SetLifespan(InitialDuration);
+    }
+
+    public override void OnDispose()
+    {
+      var amountToMove = (int)Math.Floor(UnitsToMove.Count * ((InitialDuration - Math.Floor(Duration)) / InitialDuration));
```

Comment:
Interesting, so it looks like the amounts of units that die is determined based on the remaining distance to travel. Very cool, but I feel like it'd be easier for the player to understand if it was just a flat percentage of loss if the Dreadlord is dead. What do you think?

Reply:
Not as cool but since we cant properly convey all the information to the player it is probably the right way to go about it

Reply:
For sure. Your call, I think it's a good feature either way.